### ISSUE TITLE
test footprinter default num_pins, and readonly params

### ISSUE THREAD

**Issue by ShiboSoftwareDev** [tscircuit/tscircuit.com#262]
- since num_pins should be present in all footprints, remove the "Number of Pins" input field.
- make sure to figure out which params are readonly and not display an input field for them.
Eg. If a component has a readonly pitch param, I wouldn't have the option to change it

**Comment by ShiboSoftwareDev** [comment]
Remove number of pins because you should be able to change it from the input under 'Parameters'.
Now the Parameters are sometimes readonly and sometimes are not, this depends on the their zod type created in the footprinter repo, only readonly paramters should not be editable.

**Comment by ShiboSoftwareDev** [comment]
@techmannih as I said for the input fields under 'Parameters' only some of them are readonly. For example: ![Image](https://github.com/user-attachments/assets/8d7c6b64-64c6-4fb9-974f-e5a09783bf03)
only the num_pins param is readonly(z.literal) and all the other params should be editable.

**Comment by ShiboSoftwareDev** [comment]
> [@ShiboSoftwareDev](https://github.com/ShiboSoftwareDev) pease check, Now the num_pins behaves like the other parameters and it is not editable.
> 
>  20241127125057.mp4

that's not what I said. some paramters are readonly and some are not; you'd know from their type definition. look at the old 'Number of Pins' input field to know how to handle num_pins because it is a special case, regardless of whether it is readonly or not. If you find it difficult to understand the terminology I'm using send my comments to chatgpt, give it some context and ask it to explain in simpler terms.

**Comment by ShiboSoftwareDev** [comment]
@siva222003 this issue is deprecated, should've been closed

**Comment by siva222003** [comment]
Okay, but still the chip preview breaks whenever the readonly params are changed right?
